spring-kafka/src/main/java/org/springframework/kafka/listener/ListenerContainerRegistry.java

```diff
@@ -41,6 +41,17 @@ public interface ListenerContainerRegistry {
 	@Nullable
 	MessageListenerContainer getListenerContainer(String id);
 
+	/**
+	 * Return the {@link MessageListenerContainer} with the specified id or {@code null}
+	 * if no such container exists. Returns containers that are not registered with the
+	 * registry, but exist in the application context.
+	 * @param id the id of the container
+	 * @return the container or {@code null} if no container with that id exists
+	 * @see #getListenerContainerIds()
+	 */
+	@Nullable
+	MessageListenerContainer getUnregisteredListenerContainer(String id);
```

Comment:
Why do we need this API, please?

Reply:
The old back off manager only handled containers created via `@KafkaListener`. The new one `ContainerPartitionPausingBackOffManager` will call this when it doesn't find a registered one. This seemed a reasonable place to do it (we already have `getAllListenerContainers()`, rather than restructuring the code there to handle the application context directly.